Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -0,0 +1,254 @@
+            if (null !== $nonWhiteAbove && in_array($tokens[$nonWhiteAbove]->getId(), $methodAttr, true)) {
+                $firstMethodAttrIndex = $nonWhiteAbove;
+            } else {
+                break;
+            }
+        }
+
+        if (T_DOC_COMMENT !== $tokens[$nonWhiteAbove]->getId()) {
+            $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, $nonWhiteAbove === $classStart ? 1 : 2);
+
+            return;
+        }
+
+        // there should be one linebreak between the method signature and the PHPDoc above it
+        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstMethodAttrIndex, 1);
+
+        // there should be one blank line between the PHPDoc and whatever is above
+        $nonWhiteAbovePHPDoc = $tokens->getNonWhitespaceSibling($nonWhiteAbove, -1);
+        $this->correctLineBreaks($tokens, $nonWhiteAbovePHPDoc, $nonWhiteAbove, $nonWhiteAbovePHPDoc === $classStart ? 1 : 2);
+
+        return;
+    }
+
+    private function correctLineBreaks(Tokens $tokens, $startIndex, $endIndex, $reqLineCount = 2)
+    {
+        $startIndex += 1;
+        $numbOfWhiteTokens = $endIndex - $startIndex;
+        if (0 === $numbOfWhiteTokens) {
+            $tokens->insertAt($startIndex, new Token(array(T_WHITESPACE, str_repeat("\n", $reqLineCount))));
+
+            return;
+        }
+
+        $lineBreakCount = $this->getLineBreakCount($tokens, $startIndex, $endIndex);
+        if ($reqLineCount === $lineBreakCount) {
+            return;
+        }
+
+        if ($lineBreakCount < $reqLineCount) {
+            $tokens[$startIndex]->setContent(str_repeat("\n", $reqLineCount - $lineBreakCount).$tokens[$startIndex]->getContent());
+
+            return;
+        }
+
+        // $lineCount = > $reqLineCount : check the one Token case first since this one will be true most of the time
+        if (1 === $numbOfWhiteTokens) {
+            $tokens[$startIndex]->setContent(preg_replace('/[\r\n]/', '', $tokens[$startIndex]->getContent(), $lineBreakCount - $reqLineCount));
+
+            return;
+        }
+
+        // $numbOfWhiteTokens = > 1
+        $toReplaceCount = $lineBreakCount - $reqLineCount;
+        for ($i = $startIndex; $i < $endIndex && $toReplaceCount > 0; ++$i) {
+            $tokenLineCount = substr_count($tokens[$i]->getContent(), "\n");
+            if ($tokenLineCount > 0) {
+                $tokens[$i]->setContent(preg_replace('/[\r\n]/', '', $tokens[$i]->getContent(), min($toReplaceCount, $tokenLineCount)));
+                $toReplaceCount -= $tokenLineCount;
+            }
+        }
+    }
+
+    private function getLineBreakCount(Tokens $tokens, $whiteStart, $whiteEnd)
+    {
+        $lineCount = 0;
+        for ($i = $whiteStart; $i < $whiteEnd; ++$i) {
+            $lineCount += substr_count($tokens[$i]->getContent(), "\n");
+        }
+
+        return $lineCount;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        return 55;
```

Comment:
psr0, indentation, linefeed, trailingspaces, I prop. followed one of those. I'll try to come up with some meaningful text.